cmake/modules/AddSwift.cmake

```diff
@@ -2000,15 +2000,15 @@ function(add_swift_target_library name)
             WORLD_READ)
       endif()
 
-      swift_install_in_component("${SWIFTLIB_INSTALL_IN_COMPONENT}"
+      swift_install_in_component("${SWIFTLIB_INSTALL_IN_COMPONENT}" ${name}
           FILES "${UNIVERSAL_LIBRARY_NAME}"
           DESTINATION "lib${LLVM_LIBDIR_SUFFIX}/${resource_dir}/${resource_dir_sdk_subdir}"
           PERMISSIONS ${file_permissions})
       if(sdk STREQUAL WINDOWS)
         foreach(arch ${SWIFT_SDK_WINDOWS_ARCHITECTURES})
           if(TARGET ${name}-windows-${arch}_IMPLIB)
             get_target_property(import_library ${name}-windows-${arch}_IMPLIB IMPORTED_LOCATION)
-            swift_install_in_component(${SWIFTLIB_INSTALL_IN_COMPONENT}
+            swift_install_in_component(${SWIFTLIB_INSTALL_IN_COMPONENT} ${import_library}
```

Comment:
This really should be part of the component itself.  The import library is needed to be built against, and normally `install(TARGET)` will do the work for you.